app/views/api/v2/smart_proxies/show.json.rabl

```diff
@@ -5,3 +5,13 @@ extends "api/v2/smart_proxies/main"
 node do |hostgroup|
   partial("api/v2/taxonomies/children_nodes", :object => hostgroup)
 end
+
+node(:version) { |smart_proxy| smart_proxy.statuses[:version].version['version'] }
```

Comment:
Pretty sure this still does a HTTP call over the network and can fail. IMHO that's not OK.

Reply:
As I read it this does:
https://github.com/theforeman/foreman/blob/62228620c4265f98bfe5f8eb479a915d87fc94df/app/models/smart_proxy.rb#L134
That calls this:
https://github.com/theforeman/foreman/blob/62228620c4265f98bfe5f8eb479a915d87fc94df/app/services/proxy_status/version.rb#L3-L7
Which then calls:
https://github.com/theforeman/foreman/blob/62228620c4265f98bfe5f8eb479a915d87fc94df/lib/proxy_api/version.rb#L8-L12

So that's how it does call the Foreman Proxy's HTTP API. That information is not stored in the Foreman database.